WordPress-iOS-Shared/Core/WPTextFieldTableViewCell.m

```diff
@@ -16,52 +16,54 @@ @implementation WPTextFieldTableViewCell
-        _textField.enabled = YES;
-        _textField.contentVerticalAlignment = UIControlContentVerticalAlignmentCenter;
-        _textField.delegate = self;
-        
-        _minimumLabelWidth = 90;
-        
+
         self.selectionStyle = UITableViewCellSelectionStyleNone;
         self.accessoryType = UITableViewCellAccessoryNone;
-        
-        [self.contentView addSubview:self.textField];
+
+		[self setupTextField];
     }
     
     return self;
 }
 
-- (void)layoutSubviews {
-    [super layoutSubviews];
-    
-    CGSize labelSize = [self.textLabel.text sizeWithAttributes:@{NSFontAttributeName:[UIFont boldSystemFontOfSize:17]}];
-    labelSize.width = ceil(labelSize.width/5) * 5; // Round to upper 5
-    labelSize.width = MAX(self.minimumLabelWidth, labelSize.width); // Impose alignment for shorter labels
-    CGFloat leftMargin = 0;
-    CGFloat rightMargin = self.accessoryView.frame.size.width;
-    if (!self.accessoryView && self.accessoryType != UITableViewCellAccessoryNone) {
-        rightMargin = AccessoryPadding;
-    }
-    if ([WPDeviceIdentification isiPad]) {
-        leftMargin  = iPadLeftMargin;
-        rightMargin += iPadRightMargin;
-    } else {
-        leftMargin  = iPhoneLeftMargin;
-        rightMargin += iPhoneRightMargin;
-    }
-    CGRect frame = CGRectMake(labelSize.width + leftMargin,
-                       self.textLabel.frame.origin.y,
-                       self.frame.size.width - labelSize.width - rightMargin,
-                       self.textField.frame.size.height);
-    self.textField.frame = frame;
+- (void)setupTextField
+{
+	// Use a custom textField, see below for implementation of WPCellTextField.
+	WPCellTextField *textField = [[WPCellTextField alloc] init];
+	textField.translatesAutoresizingMaskIntoConstraints = NO;
+	textField.adjustsFontSizeToFitWidth = YES;
+	textField.textColor = [UIColor blackColor];
+	textField.backgroundColor = [UIColor clearColor];
+	textField.autocorrectionType = UITextAutocorrectionTypeNo;
+	textField.autocapitalizationType = UITextAutocapitalizationTypeNone;
+	textField.textAlignment = NSTextAlignmentLeft;
+	textField.clearButtonMode = UITextFieldViewModeNever;
+	textField.enabled = YES;
+	textField.contentVerticalAlignment = UIControlContentVerticalAlignmentCenter;
+	textField.delegate = self;
+
+	[self.contentView addSubview:textField];
+	UILayoutGuide *layoutGuide = self.contentView.readableContentGuide;
+	[NSLayoutConstraint activateConstraints:@[
+											  [textField.leadingAnchor constraintEqualToAnchor:layoutGuide.leadingAnchor],
+											  [textField.trailingAnchor constraintEqualToAnchor:layoutGuide.trailingAnchor],
+											  [textField.topAnchor constraintEqualToAnchor:self.contentView.topAnchor],
+											  [textField.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor],
+											  [textField.centerYAnchor constraintEqualToAnchor:self.contentView.centerYAnchor]
```

Comment:
Do you need the center Y anchor if you're already pinning to the top and bottom?